avocado/plugins/assets.py

```diff
@@ -0,0 +1,245 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+"""
+Assets subcommand
+"""
+
+import ast
+import os
+
+from avocado.core import data_dir
+from avocado.core import safeloader
+
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+
+from avocado.utils import asset
+from avocado.utils import data_structures
+
+
+class FetchAssetHandler(ast.NodeVisitor):  # pylint: disable=R0902
+    """
+    Handles the parse of instrumented tests for `fetch_asset` statements
+    and fetch it.
+    """
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.pattern = 'fetch_asset'
+        self.asgmts = {}
+        self.calls = []
+
+        # hold current class and current method to make sure we have the
+        # correct context for the assignment statement.
+        # at this time, module constants and class attributes are discarded.
+        self.current_klass = None
+        self.current_method = None
+
+        # check if we have valid instrumented tests
+        # discards disabled tests
+        self.tests = safeloader.find_avocado_tests(self.file_name)[0]
+
+        # create Abstract Syntax Tree from test source file
+        with open(self.file_name) as source_file:
+            self.tree = ast.parse(source_file.read(), self.file_name)
+
+        # build list of keyword arguments from calls that match pattern
+        self.visit(self.tree)
+    # __init__()
```

Comment:
The closing of methods/classes with comments is not present anywhere else in Avocado.  It's not wrong, but it's a good idea to conform to the project style as much as possible.  In short, I'd drop these.